sdk/Sdk.Generators/KnownTypes.cs

```diff
@@ -14,12 +18,34 @@ namespace Microsoft.Azure.Functions.Worker.Sdk.Generators
         internal readonly INamedTypeSymbol ValueTaskType;
         internal readonly INamedTypeSymbol ValueTaskOfTTypeOpt;
 
+        internal readonly INamedTypeSymbol IEnumerable;
+        internal readonly INamedTypeSymbol IEnumerableGeneric; // IEnumerable<T>
+        internal readonly INamedTypeSymbol IEnumerableOfKeyValuePair;
+        internal readonly INamedTypeSymbol StringType;
+        internal readonly INamedTypeSymbol ByteArray;
+        internal readonly INamedTypeSymbol ByteType;
+        internal readonly INamedTypeSymbol VoidType;
+        internal readonly INamedTypeSymbol ReadOnlyMemoryOfBytes;
+        internal readonly INamedTypeSymbol LookupGeneric;
+        internal readonly INamedTypeSymbol DictionaryGeneric;
+
         internal KnownTypes(Compilation compilation)
         {
             TaskType = compilation.GetTypeByMetadataName(typeof(Task).FullName)!;
             TaskOfTType = compilation.GetTypeByMetadataName(typeof(Task<>).FullName)!;
             ValueTaskType = compilation.GetTypeByMetadataName(typeof(ValueTask).FullName)!;
             ValueTaskOfTTypeOpt = compilation.GetTypeByMetadataName(typeof(ValueTask<>).FullName)!;
+
+            IEnumerable = compilation.GetTypeByMetadataName(typeof(IEnumerable).FullName)!;
```

Comment:
I would slightly prefer all the fields/properties here to be lazily initialized. This way we only read from `compilation` the types we actually use as the generator executes.